### PR TITLE
Fix issue with logger due to iteration over sys.modules.items()

### PR DESCRIPTION
**Pull request by astrofrog**
In rare cases the iteration over sys.modules.items() can fail with the error:

```
dictionary changed size during iteration
```

so we take a copy to avoid this. This is the same error as happened in this package:

https://bitbucket.org/cherrypy/cherrypy/issue/1280/autoreloader-throws-dictionary-changed

If there are no objections, I will merge by Monday to make sure it gets into 1.0.

### CODE REVIEW

**Comment by astrofrog** [comment]
This should be uncontroversial, so merging so that it makes 1.0.

File: astropy/logger.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def _showwarning(self, *args, **kwargs):
                     mod_name = mod.__name__
                     break
         else:  # pragma: py2
-            for name, mod in sys.modules.items():
+            for name, mod in list(sys.modules.items()):
```
Comment (mdboom) [inline astropy/logger.py]: This is fine, but shouldn't be necessary on Python 2 where `items` already returns a copied list of items.

Comment (astrofrog) [inline astropy/logger.py]: Right - would it make sense to write a ``get_modules`` function that avoids the copy in Python 2?

Comment (embray) [inline astropy/logger.py]: If it doesn't already, I feel like `six` should have `keys()`, `values()`, and `items()` functions (just as it has `iterkeys()`, `itervalues()`, and `iteritems()` functions) that reproduce the Python 2 behavior for those `dict` methods.

**Comment by embray** [comment]
This is pretty obscure, but okay, good to know.

File: astropy/logger.py (hunk 1 above)
Comment (astrofrog) [inline astropy/logger.py]: I agree - maybe I can try and do a pull request to ``six`` to see if they'd accept those changes. In the mean time I can add a helper function for the logger specifically.

Comment (embray) [inline astropy/logger.py]: I wouldn't bother with the latter.

**Pull request merged**